$OpenBSD: patch-salt_grains_core_py,v 1.4 2020/08/03 13:30:06 jasper Exp $

Unbreak minion without swap configured:

  File "/usr/local/lib/python3.8/site-packages/salt/grains/core.py", line 534, in _bsd_memdata
      grains["swap_total"] = int(swap_total) // 1024 // 1024
      ValueError: invalid literal for int() with base 10: 'no'

Index: salt/grains/core.py
--- salt/grains/core.py.orig
+++ salt/grains/core.py
@@ -525,7 +525,7 @@ def _bsd_memdata(osdata):
         if osdata["kernel"] in ["OpenBSD", "NetBSD"]:
             swapctl = salt.utils.path.which("swapctl")
             swap_data = __salt__["cmd.run"]("{0} -sk".format(swapctl))
-            if swap_data == "no swap devices configured":
+            if re.match(r"(swapctl: )?no swap devices configured", swap_data):
                 swap_total = 0
             else:
                 swap_total = swap_data.split(" ")[1]
